frontend/packages/console-shared/src/components/markdown-extensions/MarkdownExecuteSnippet.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import useCloudShellAvailable from '@console/app/src/components/cloud-shell/useCloudShellAvailable';
+import { useCloudShellCommandDispatch } from '@console/app/src/redux/actions/cloud-shell-dispatchers';
+import { useEventListener } from '../../hooks';
+import { Tooltip } from '../Tooltip/Tooltip';
+import { MARKDOWN_EXECUTE_BUTTON_ID, MARKDOWN_SNIPPET_ID } from './const';
+
+type ExecuteSnippetProps = {
+  element: HTMLElement;
+  rootSelector: string;
+  docContext: HTMLDocument;
+};
+
+const ExecuteSnippet: React.FC<ExecuteSnippetProps> = ({ element, rootSelector, docContext }) => {
+  const setCloudShellCommand = useCloudShellCommandDispatch();
+  const [showRunning, setShowRunning] = React.useState<boolean>(false);
+  const executeTextId = element.getAttribute(MARKDOWN_EXECUTE_BUTTON_ID);
+  const textToExecute = (docContext.querySelector(
+    `${rootSelector} [${MARKDOWN_SNIPPET_ID}="${executeTextId}"]`,
+  ) as HTMLElement).innerText;
+
+  useEventListener(
+    element,
+    'click',
+    React.useCallback(() => {
+      setCloudShellCommand(textToExecute);
+      setShowRunning(true);
+      element.setAttribute('data-executed', '');
+    }, [textToExecute, element, setCloudShellCommand]),
+  );
+
+  return (
+    <Tooltip
+      reference={() => element}
+      onShow={() => {
+        element.removeAttribute('data-executed');
+      }}
+      onHide={() => {
+        setShowRunning(false);
+      }}
+    >
+      {showRunning ? 'Running in Web Terminal' : 'Run in Web Terminal'}
+    </Tooltip>
+  );
+};
+
+type MarkdownExecuteCommandProps = {
+  docContext: HTMLDocument;
+  rootSelector: string;
+};
+
+const MarkdownExecuteSnippet: React.FC<MarkdownExecuteCommandProps> = ({
+  docContext,
+  rootSelector,
+}) => {
+  const elements = docContext.querySelectorAll(`${rootSelector} [${MARKDOWN_EXECUTE_BUTTON_ID}]`);
```

Comment:
Use memo so that you don't query in render flow.

Reply:
@sahil143 Seems like you missed to resolve this one.

Reply:
Sorry missed this one. I tried using the useMemo for this but the elements are not present if I use the useMemo here because the first render document doesn't have the queried elements whereas it is present in the next renders but since the dependency doesn't know that document is changed useMemo is not triggered.

Reply:
@sahil143 please create an issue to track this. The render extensions must be rendered only once the content is present on the page. This must be a bug from when we changed from iframe to inline content. 
We should investigate.